openedx/adg/lms/applications/managers.py

```diff
@@ -23,29 +23,47 @@ class MultilingualCourseGroupManager(Manager):
     Manager for MultilingualCourseGroup
     """
 
+    def common_course_groups(self):
+        return self.get_queryset().filter(
+            is_common_business_line_prerequisite=False, business_line_prerequisite__isnull=True
+        )
+
     def prereq_course_groups(self):
         """
         Get non-empty prerequisite course groups
         """
         return self.get_queryset().filter(is_program_prerequisite=True, multilingual_courses__isnull=False).distinct()
 
-    def get_courses(self, user, is_prereq=False):
+    def business_line_course_groups(self, user):
+        return self.get_queryset().filter(
+            Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite=user.application.business_line)
+        )
+
+    def get_courses(self, user, is_prereq=False, business_line_courses=False):
         """
-        Get courses from course groups.
+        Get course groups for a user
 
         Args:
-            is_prereq (bool):  List of MultilingualCourseGroups
-            user (User): user for which we need to find courses
+            user (User): User for which we need to find courses
+            is_prereq (bool): Bool to check that we only need to return prerequisite courses
+            business_line_courses (bool): Bool to check that we only need to return business line prerequisites
 
         Returns:
             list: List of courses which contains a course from each group
         """
-        courses_list = []
-        course_groups = self.prereq_course_groups() if is_prereq else self.get_queryset()
+        course_groups = self.common_course_groups()
+
+        if is_prereq:
+            course_groups = self.prereq_course_groups()
+
+        if business_line_courses:
+            course_groups = self.business_line_course_groups(user)
 
+        courses_list = []
```

Comment:
How about writing separate getters for all these three kind of course groups and moving the code below to a utility and utilizing in all three getters?